include/uapi/sound/sof-ipc.h

```diff
@@ -451,7 +451,8 @@ struct sof_ipc_stream_params {
 	enum sof_ipc_stream_direction direction;
 	enum sof_ipc_frame frame_fmt;
 	enum sof_ipc_buffer_format buffer_fmt;
-	uint32_t stream_tag;
+	uint32_t host_dma_ch;
+	uint32_t be_dma_ch;
```

Comment:
This doesn't seem quite right. This solution will only work if you have a passthrough pipeline. But for the general case you can have multiple Host streams mixed into one link DMA for playback, and conversely for capture you can have multiple link DMAs that are muxed into a single host DMA. So we should use a different IPC to set the link DMA configurations. We should view the link DMA as a DAI really.

Reply:
yes, I understand your idea, a big picture and I also investigated these cases. I think it is a feature development not a bug fixing for current status. If you want a solution to accommodate all these cases, we may need a discussing in our team to get a complete solution.